src/controller/helpers.js

```diff
@@ -14,38 +14,55 @@ class auth {
     return bcrypt.compareSync(password, hashPassword);
   }
 
-  static generateToken(id) {
+  static generateToken(id, isadmin) {
     const token = jwt.sign({
       userId: id,
+      isAdmin: isadmin,
     },
     process.env.SECRET, { expiresIn: '1d' });
     return token;
   }
 
-  static async tokenHandler(table, req, res, next) {
+  static async verifyToken(req, res, next) {
     const token = req.headers.tokens;
     if (!token) {
       return errorHandler(400, res, 'Token is not provided');
     }
     try {
       const decoded = await jwt.verify(token, process.env.SECRET);
-      const { rows } = await db.query(queries.selectById(table, 'id', decoded.userId));
+      if (decoded.isAdmin === 'true') {
+        return errorHandler(403, res, 'Token Forbidden');
+      }
+      const { rows } = await db.query(queries.selectById('users', 'id', decoded.userId));
       if (!rows[0]) {
-        return errorHandler(400, res, 'The token you provided is invalid');
+        return errorHandler(401, res, 'Invalid token');
       }
-      req.user = { id: decoded.userId };
-      next();
+      req.user = { id: decoded.userId, isadmin: decoded.isAdmin };
+      return next();
     } catch (error) {
       return errorHandler(400, res, error);
     }
   }
 
-  static async verifyToken(req, res, next) {
-    auth.tokenHandler('users', req, res, next);
-  }
-
   static async adminVerifyToken(req, res, next) {
-    auth.tokenHandler('admins', req, res, next);
+    const token = req.headers.tokens;
+    if (!token) {
+      return errorHandler(400, res, 'Token is not provided');
```

Comment:
Do you think that error code 400 is appropriate for this scenario?

Reply:
I believe it should have been 401, authentication error

Reply:
Take note and incorporate it where ever it applies.